tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp

```diff
@@ -848,19 +849,31 @@ std::string SwiftLangSupport::resolvePathSymlinks(StringRef FilePath) {
 
   return InputPath;
 #else
-  char full_path[MAX_PATH];
-
-  HANDLE fileHandle = CreateFileA(
-      InputPath.c_str(), GENERIC_READ, 0, nullptr, OPEN_EXISTING,
-      FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OPEN_REPARSE_POINT, nullptr);
+  wchar_t full_path[MAX_PATH] = {0};
+  llvm::SmallVector<llvm::UTF16, 50> utf16Path;
+  llvm::convertUTF8ToUTF16String(InputPath.c_str(), utf16Path);
+
+  HANDLE fileHandle = CreateFileW(
+      (LPCWSTR)utf16Path.data(), 
+      0, 
+      FILE_SHARE_READ | FILE_SHARE_WRITE, 
+      nullptr, 
+      OPEN_EXISTING,
+      FILE_FLAG_BACKUP_SEMANTICS,
+      nullptr);
 
   if (fileHandle == INVALID_HANDLE_VALUE)
     return InputPath;
 
-  DWORD success = GetFinalPathNameByHandleA(
-      fileHandle, full_path, sizeof(full_path), FILE_NAME_NORMALIZED);
+  DWORD getPathSuccess = GetFinalPathNameByHandleW(
```

Comment:
Why the rename of `success`?

Reply:
I added a `convertSuccess` check and it seemed weird to have a `success` and `convertSuccess`.

Reply:
You could just do `success &= ...` ;-)